## September 18th 2015
10:00am @Home

1. Review [Genetic Science Learning Labs](http://learn.genetics.utah.edu/content/labs/extraction/) 
   * DNA Extraction

2. Background: Annotation of Genes
   * Annotation is the process by which pertinent information about these raw DNA sequences is added to the genome databases. This involves describing different regions of the code and identifying which regions can be called genes.
   * http://www.plantgdb.org/tutorial/annotatemodule/
   
3. Background: Research Articles
   * http://www.genomebiology.com/2014/15/3/R59

## September 20th 2015
6:00PM @Home

1. http://weatherby.genetics.utah.edu/MAKER/wiki/index.php/MAKER_Tutorial
   * Annotations are descriptions of different features of the genome, and they can be structural or functional in        nature. Examples: Structural Annotations: exons, introns, UTRs, splice forms. Functional Annotations: process a      gene is involved in (metabolism), molecular function (hydrolase), location of expression (expressed in the    mitochondria), etc. 

2. Genome sequence itself is not very useful. The first question that occurs to most of us when a genome is sequenced is, "where are the genes?" To **identify the genes we need to annotate the genome**. And while most researchers probably don't give annotations a lot of thought, they use them everyday.
3. Examples of Annotation Databases: Ensembl, RefSeq, FlyBase, WormBase, Mouse Genome Informatics
   Every time we use techniques such as RNAi, PCR, gene expression arrays, targeted gene knockout, or ChIP we are   **basing our experiments on the information derived from a digitally stored genome annotation**. If an annotation is correct, then these experiments should succeed; however, if an annotation is incorrect then the experiments that are based on that annotation are bound to fail. 
